File: frontend/static/js/modules/state/engine_state.js
export const engineState = {
    score: 0,
    depth: 0,
    nodes: 0,
    nps: 0,
    best_move: "--",
    pv: [],
    multiPv: [],
    is_thinking: false,
    status: "IDLE"
};

export function updateEngineState(payload) {
    engineState.score = payload.score ?? engineState.score;
    engineState.depth = payload.depth ?? engineState.depth;
    engineState.nodes = payload.nodes ?? engineState.nodes;
    engineState.nps = payload.nps ?? engineState.nps;
    engineState.best_move = payload.best_move ?? payload.move ?? engineState.best_move;
    engineState.pv = payload.pv ?? engineState.pv;
    engineState.multiPv = payload.multiPv ?? payload.multi_pv ?? engineState.multiPv;
    engineState.is_thinking = payload.is_thinking ?? engineState.is_thinking;
    engineState.status = payload.status ?? engineState.status;
}

File: backend/state/store/models/engine_state.py
from dataclasses import dataclass, field
from typing import List, Dict, Any, Optional

@dataclass(frozen=True)
class EngineState:
    bestmove: Optional[str] = None
    score: float = 0.0
    depth: int = 0
    nodes: int = 0
    nps: int = 0
    pv: List[str] = field(default_factory=list)
    multipv: List[Dict[str, Any]] = field(default_factory=list)
    engine_status: str = "READY"
    is_thinking: bool = False
